Improve documentation for X-Opaque-ID (#46167)
elastic/elasticsearch: docs/reference/index-modules/slowlog.asciidoc

logger.index_search_slowlog_rolling.additivity = false
--------------------------------------------------

> ==== Identifying search slow log origin

Reviewer: ```suggestion
[float]
==== Identifying search slow log origin
```

> It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then this value

Reviewer: ```suggestion
It is often useful to identify what triggered a slow running query. If a call was initiated with an `X-Opaque-ID` header, then the user ID
```

> will be present in Search Slow logs as an additional **id** field (scroll to the right).

Reviewer: ```suggestion
is included in Search Slow logs as an additional **id** field (scroll to the right).
```

> ```

Reviewer: I'd rewrite this to use the Asciidoc source code blocks.

For example:
```
[source,txt]
----
[2019-08-30T11:59:37,786][WARN ][i.s.s.query              ] [node-0] [index6][0] took[78.4micros], took_millis[0], total_hits[0 hits], stats[], search_type[QUERY_THEN_FETCH], total_shards[1], source[{"query":{"match_all":{"boost":1.0}}}], id[MY_USER_ID],
----
```

Also consider replacing `[2019-08-30T11:59:37,786]` with a more future-proof timestamp like `[2030-08-30T11:59:37,786]`

[2019-08-30T11:59:37,786][WARN ][i.s.s.query              ] [node-0] [index6][0] took[78.4micros], took_millis[0], total_hits[0 hits], stats[], search_type[QUERY_THEN_FETCH], total_shards[1], source[{"query":{"match_all":{"boost":1.0}}}], id[MY_USER_ID],
```

> the same applies to JSON logs

Reviewer: ```suggestion
The user ID is also included in JSON logs.
```

> ```

Reviewer: Also replace with a code block.
```
[source,js]
----
{
  "type": "index_search_slowlog",
  "timestamp": "2019-08-30T11:59:37,786+02:00",
  "level": "WARN",
  "component": "i.s.s.query",
  "cluster.name": "distribution_run",
  "node.name": "node-0",
  "message": "[index6][0]",
  "took": "78.4micros",
  "took_millis": "0",
  "total_hits": "0 hits",
  "stats": "[]",
  "search_type": "QUERY_THEN_FETCH",
  "total_shards": "1",
  "source": "{\"query\":{\"match_all\":{\"boost\":1.0}}}",
  "id": "MY_USER_ID",
  "cluster.uuid": "Aq-c-PAeQiK3tfBYtig9Bw",
  "node.id": "D7fUYfnfTLa2D7y-xw6tZg"
}
----
// NOTCONSOLE
```
